### PR TITLE
feat(linter): add `radix` rule

### PR DESCRIPTION
**Pull request by KubaJastrz**

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/radix.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    /// ### Example
+    /// ```javascript
+    /// // error
+    /// var num = parseInt("071");      // 57
+    ///
+    /// // success
+    /// var num = parseInt("071", 10);  // 71
+    /// ```
+    Radix,
+    correctness
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/radix.rs:59]: I think this rule should be `pedantic`.

Comment (KubaJastrz) [reply crates/oxc_linter/src/rules/eslint/radix.rs]: Changed

File: crates/oxc_linter/src/rules/eslint/radix.rs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        ("class C { #parseInt; foo() { Number.#parseInt(foo, 'bar'); } }", None),
+        ("class C { #parseInt; foo() { Number.#parseInt(foo, 10); } }", Some(json!(["as-needed"]))),
+        ("var parseInt; parseInt();", None),
+        ("var parseInt; parseInt(foo);", Some(json!(["always"]))),
+        ("var parseInt; parseInt(foo, 10);", Some(json!(["as-needed"]))),
+        ("var Number; Number.parseInt();", None),
+        ("var Number; Number.parseInt(foo);", Some(json!(["always"]))),
+        ("var Number; Number.parseInt(foo, 10);", Some(json!(["as-needed"]))),
+        // ("/* globals parseInt:off */ parseInt(foo);", Some(json!(["always"]))),
+        // ("Number.parseInt(foo, 10);", Some(json!(["as-needed"]))), // { globals: { Number: "off" } }
```
Comment (KubaJastrz) [inline crates/oxc_linter/src/rules/eslint/radix.rs:196]: Do we care about these? The only rule that kinda checks for globals is `no_undef`, but even that one has a bunch of commented out tests.

Comment (Boshen) [reply crates/oxc_linter/src/rules/eslint/radix.rs:196]: We can leave these out for now.

**Review by Boshen — APPROVED** [review]

**Comment by Boshen** [comment]
```
error: binary comparison to literal `Option::None`
  --> crates/oxc_linter/src/rules/eslint/radix.rs:77:24
   |
77 |                     if ctx.symbols().get_symbol_id_from_name("parseInt") == None {
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `Option::is_none()` instead: `ctx.symbols().get_symbol_id_from_name("parseInt").is_none()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#partialeq_to_none
   = note: `-D clippy::partialeq-to-none` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::partialeq_to_none)]`
```

quick clippy fix.

run `just lint`

**Comment by Boshen** [comment]
Thank you!

**Pull request merged**